core/componentConfig.go

```diff
@@ -111,6 +128,26 @@ func (cc *ComponentConfig) GetSubcomponentConfig(subcomponentPath []string) (sub
 	return subcomponentConfig
 }
 
+func (cc *ComponentConfig) HasSubcomponentConfig(subcomponentPath []string) (hasSubComponentconfig bool) {
```

Comment:
`hasSubComponentconfig` is never used. Lets remove it (just keep the `bool` return type)